bindings/python/tests/test_parse.py

```diff
@@ -0,0 +1,237 @@
+from context import scripts
+import scripts.parse as parse
+
+import json
+import os
+
+import clang.cindex as clang
+
+
+def create_compilation_database(tmp_path, filepath):
+    input = tmp_path / "compile_commands.json"
+    x = [
+        {
+            "directory": f"{tmp_path}",
+            "command": f"/usr/bin/clang++ -std=c++14 {filepath}",
+            "file": f"{filepath}",
+        }
+    ]
+    input.write_bytes(str(x).encode())
+    tmp_path = str(tmp_path)
+    return tmp_path
+
+
+def get_parsed_info(tmp_path, file_contents):
+    source_path = tmp_path / "file.hpp"
+    source_path.write_bytes(str(file_contents).encode())
+
+    parsed_info = parse.parse_file(
+        source=str(source_path),
+        compilation_database_path=create_compilation_database(
+            tmp_path=tmp_path, filepath=source_path
+        ),
+    )
+
+    return parsed_info
+
+
+def test_types(tmp_path):
+    file_contents = ""
+    parsed_info = get_parsed_info(tmp_path=tmp_path, file_contents=file_contents)
+
+    assert type(parsed_info) is dict
+    assert type(parsed_info["members"]) is list
+
+
+def test_translation_unit(tmp_path):
+    file_contents = ""
+    parsed_info = get_parsed_info(tmp_path=tmp_path, file_contents=file_contents)
+
+    assert parsed_info["kind"] == "TRANSLATION_UNIT"
+
+
+def test_namespace(tmp_path):
+    file_contents = "namespace Anamespace {}"
+    parsed_info = get_parsed_info(tmp_path=tmp_path, file_contents=file_contents)
+
+    namespace = parsed_info["members"][0]
+
+    assert namespace["kind"] == "NAMESPACE"
```

Comment:
Also test the names, for the complete parse check

Reply:
Done :+1: